docs/v3-ea-components/transport-types/websocket-transport.md

```diff
@@ -1,3 +1,227 @@
 # Websocket Transport
 
-_Guide coming soon_
+WebSocketTransport is used to fetch data from a Provider using websocket protocol. In order to use WebSocketTransport you need to provide a `url` for connection and handler for  `message` event. Additionally you can also provide options for connection, handler for `open` event and builders to customize subscribe and unsubscribe messages.
+
+Example usage of WebSocketTransport
+```typescript  
+ export const wsTransport = new WebSocketTransport<EndpointTypes>({
+      url: (context, desiredSubs) => context.adapterConfig.WS_API_ENDPOINT,
+      handlers: {
+        message(message){
+          return [
+            {
+              params: {
+                base: message.data.base,
+                quote:  message.data.quote,
+              },
+              response: {
+                result: message.data.price,
+                data: {
+                  result: message.data.price,
+                },
+                timestamps: {
+                  providerIndicatedTimeUnixMs: Math.round(message.data.ts / 1e6),
+                },
+              },
+            },
+          ]
+        },
+      },
+      builders: {
+        subscribeMessage: (params) => ({
+          action: 'subscribe',
+          symbols: [`${params.base}.${params.quote}`],
+        }),
+        unsubscribeMessage: (params) => ({
+          action: 'unsubscribe',
+          symbols: [`${params.base}.${params.quote}`],
+        }),
+      },
+    })
+```
+### Connecting to Data Provider
+**url**  method takes the adapter context and the desired subscriptions and should return connection url as a string or a Promise of string.
+Some Data Providers support subscribing to feeds directly using url, without separate message. In that case second argument of the method can be used to construct connection/subscription url.
+```typescript
+    url: (context, desiredSubs) => {
+        const params = desiredSubs.map((sub) => `${sub.base}-${sub.quote}`).join(',')
+        return `${context.adapterConfig.WS_API_ENDPOINT}/${params}`
+      },
+```
+In some cases (authentication for example) it is necessary to pass headers as well. `options` method can be used to pass additional information for connection.
+```typescript
+      url: (context) => context.adapterConfig.WS_API_ENDPOINT,
+      options: async (context) => ({
+        headers: {
+          'x-auth-token': context.adapterConfig.API_KEY
+        },
+      })
+```
+
+### Handling opened connection (optional)
+**open** method of **handlers** object is executed once the websocket connection is successfully opened. In some scenarios it is necessary to perform logic when connecting to websocket but before sending subscription messages. It accepts raw websocket connection and adapter context as arguments and should return a Promise that either resolves or rejects with error message
```

Comment:
```suggestion
The **open** method of the **handlers** object is executed once the websocket connection is successfully opened. In some scenarios it is necessary to perform logic when connecting to websocket, but before sending subscription messages. It accepts a raw websocket connection and adapter context as arguments and should return a Promise that either resolves or rejects with error message
```

> return a Promise that either resolves

Resolves with what?

Reply:
with anything or nothing. The important thing is that it resolves.

Reply:
Got it, I think it's fine then